src/Request/Files.php

```diff
@@ -69,9 +69,11 @@ protected function init($src, &$tgt)
             }
         } else {
             // not a target, create sub-elements and init them too
-            foreach ($src as $key => $val) {
-                $tgt[$key] = array();
-                $this->init($val, $tgt[$key]);
```

Comment:
I think probably the best way is to check is_array before this init method.

Reply:
Not sure if I follow you, you mean like this?

```
$tgt[$key] = array();
if (is_array($tgt[$key])
    $this->init($val, $tgt[$key]);
```

But that will always be an array because of the previous line: `$tgt[$key] = array();`?

Reply:
foreach is on the first argument passed. That means it is $val should be an array .